xgress/link_send_buffer.go

```diff
@@ -267,6 +266,12 @@ func (buffer *LinkSendBuffer) run() {
 	}
 }
 
+func (buffer *LinkSendBuffer) checkForCloseOnEmpty() {
+	if buffer.closeWhenEmpty.Load() && len(buffer.buffer) == 0 && !buffer.x.Closed() && buffer.x.IsEndOfCircuitSent() {
+		go buffer.x.Close()
```

Comment:
Does this race? I don't know from the PR perspective if x.Close() has internal locking or if we just give up on double close and move on.

Reply:
It's got an atomic guard